go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -433,14 +403,14 @@ func (oc *SecondaryLayer3NetworkController) addUpdateNodeEvent(node *kapi.Node,
 }
 
 func (oc *SecondaryLayer3NetworkController) addNode(node *kapi.Node) ([]*net.IPNet, error) {
-	hostSubnets, err := oc.allocateNodeSubnets(node, oc.masterSubnetAllocator)
-	if err != nil {
-		return nil, err
-	}
-
-	hostSubnetsMap := map[string][]*net.IPNet{oc.GetNetworkName(): hostSubnets}
-	err = oc.UpdateNodeHostSubnetAnnotationWithRetry(node.Name, hostSubnetsMap, nil)
-	if err != nil {
+	// Node subnet for the secondary layer3 network is allocated by cluster manager.
+	// Make sure that the node is allocated with the subnet before proceeding
+	// to create OVN Northbound resources.
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, oc.GetNetworkName())
+	if err != nil && !util.IsAnnotationNotSetError(err) || len(hostSubnets) < 1 {
+		// Log the error and return. Retry framework will keep retrying
+		// until cluster manager allocates the node subnet for the secondary network.
+		klog.Infof("Failed to get node %s host subnets annotations: %v", node.Name, err)
```

Comment:
Just return the error. The caller will log.

I think you need to return an error on every case, otherwise there won't be a retry:
* So no `!util.IsAnnotationNotSetError(err)` check
* Retrun error if `len(hostSubnets) < 1`

Reply:
Ack